zc_events/client.py

```diff
@@ -238,11 +240,40 @@ def send_email(self, *args, **kwargs):
             with attachments {} and files {}'''
             logger.info(msg.format(email_uuid, to, from_email, attachments, files))
 
-        event_data = generate_email_data(*args, **kwargs)
+        event_data = generate_email_data(email_uuid, *args, **kwargs)
 
         if logger:
             logger.info('MICROSERVICE_SEND_EMAIL: Sent email with UUID {} and data {}'.format(
                 email_uuid, event_data
             ))
 
         self.emit_microservice_event('send_email', **event_data)
+
+    def emit_index_rebuild_event(self, event_name, resource_type, model, batch_size, serializer, queryset=None):
```

Comment:
This is a very long running process, especially for large tables. Having long processes in the workers has a lot of down sides. From what I can see this loops through all instances in the model, serializes them, pushes them to S3. For a large table this could take hours!

My suggestion is to break this down into multiple methods that could run on different workers.